tests/bsim/bluetooth/audio/src/bap_scan_delegator_test.c

```diff
@@ -5,76 +5,830 @@
  */
 
 #ifdef CONFIG_BT_BAP_SCAN_DELEGATOR
+#include <zephyr/bluetooth/bluetooth.h>
+#include <zephyr/bluetooth/audio/audio.h>
 #include <zephyr/bluetooth/audio/bap.h>
+#include <zephyr/sys/byteorder.h>
 #include "common.h"
 
 extern enum bst_result_t bst_result;
 
+#define SYNC_RETRY_COUNT          6 /* similar to retries for connections */
+#define PA_SYNC_SKIP              5
+
 static volatile bool g_cb;
 static volatile bool g_pa_synced;
+CREATE_FLAG(flag_broadcast_code_received);
```

Comment:
If we start to use CREATE_FLAG() isn't it a good idea to change existing "flags" in the same way? Or would this require too much work and another PR?

Reply:
Good question. IMO it's not a problem to use both, even if we should ideally use `CREATE_FLAG`. I can update the test to only use `CREATE_FLAG` if you want

Reply:
I think we should stick to one way of doing this just for readability, and I think it's better to do this here, as later this can be forgotten.

Reply:
Added as 2 new commits